app/org/sagebionetworks/bridge/dynamodb/DynamoUploadFieldDefinition.java

```diff
@@ -0,0 +1,90 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import javax.annotation.Nonnull;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.upload.UploadFieldDefinition;
+import org.sagebionetworks.bridge.models.upload.UploadFieldType;
+import org.sagebionetworks.bridge.validators.Validate;
+
+/**
+ * Dynamo DB implementation of UploadFieldDefinition. While there is nothing specific to Dynamo DB in this class, this
+ * class exists to distinguish itself from potential other implementations.
+ */
+@JsonDeserialize(builder = DynamoUploadFieldDefinition.Builder.class)
+public class DynamoUploadFieldDefinition implements UploadFieldDefinition {
+    private final @Nonnull String name;
```

Comment:
This actually comes from the Google implementation of JSR-305:
https://code.google.com/p/jsr-305/source/browse/trunk/ri/src/main/java/javax/annotation/Nonnull.java

It has 3 benefits:
1. It provides documentation about the preconditions of a given method or class.
2. Static code analysis tools (such as IntelliJ, FindBugs, and possibly Eclipse) can tell us if our preconditions are violated.
3. There are tools that can validate objects using annotations.

It looks like there's a separate annotation coming in Java 8. If/when we migrate, we can consider migrating all of our annotations to the Java 8 version.